From 4d2c5fa831ac73cac2557a024b327e71f95a5814 Mon Sep 17 00:00:00 2001
From: Asish Bhattacharya <asishb@codeaurora.org>
Date: Tue, 8 Dec 2020 10:48:15 +0530
Subject: [PATCH] ALSA: uapi: add check to avoid duplicate include of time.h in
 asound.h

The uapi header asound.h defines types based on struct timespec. We need
to #include <time.h> to get access to the definition of this struct.

Glibc version of stdlib.h indirectly includes time.h. This results in
duplicate  include as time.h is included from asound.h and stdlib.h both.

Add check to avoid duplicate include from stdlib.h and asound.h

Change-Id: Ia55e01e17270c480e79c56ce648182e706fc668d
Signed-off-by: Asish Bhattacharya <asishb@codeaurora.org>
---
 include/uapi/sound/asound.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/uapi/sound/asound.h b/include/uapi/sound/asound.h
index bfcb2d6..6854bfd 100644
--- a/include/uapi/sound/asound.h
+++ b/include/uapi/sound/asound.h
@@ -26,7 +26,10 @@
 
 #if defined(__KERNEL__) || defined(__linux__)
 #include <linux/types.h>
+#if !defined(_STRUCT_TIMESPEC) && !defined(__timespec_defined)
 #include <linux/time.h>
+#define __timespec_defined 1
+#endif
 #else
 #include <sys/ioctl.h>
 #endif
-- 
2.7.4

